sambacc/rados_opener.py

```diff
@@ -20,58 +20,80 @@
 
 import io
 import json
+import logging
 import typing
 import urllib.request
 
 from . import url_opener
 from .typelets import ExcType, ExcValue, ExcTraceback
 
 _RADOSModule = typing.Any
+_RADOSObject = typing.Any
 
 _CHUNK_SIZE = 4 * 1024
 
+_logger = logging.getLogger(__name__)
+
 
 class RADOSUnsupported(Exception):
     pass
 
 
+class _RADOSInterface:
+    api: _RADOSModule
+    client_name: str
+    full_name: bool
```

Comment:
Why not have `__init__` function with explicit types? IMHO it would make the code more readable.